staging/mausezahn.c

```diff
@@ -684,18 +684,49 @@ int getopts (int argc, char *argv[])
 				(!ipv6_mode && get_ip_range_src(tx.ip_src_txt))
 				)
 			{
-				// name2addr4 accepts a DOTTED DECIMAL ADDRESS or a FQDN:
-				if (ipv6_mode)
+				// name2addr{4,6} accept a valid IP address or a FQDN:
+				if (ipv6_mode) {
 					tx.ip6_src = libnet_name2addr6 (l, tx.ip_src_txt, LIBNET_RESOLVE);
-				else
-					tx.ip_src = libnet_name2addr4 (l, tx.ip_src_txt, LIBNET_RESOLVE);
+					if (libnet_in6_is_error(tx.ip6_src)) {
+						/* libnet_in6_is_error returns 1 for the valid IPv6 address
+						* ffff:ffff:ffff:ffff:ffff:ffff. Use an additional inet_pton()
+						* check to cover cases where this address is specified
+						* as source
+						*/
+						struct in6_addr src_check;
+						if (inet_pton(AF_INET6, tx.ip_src_txt, &src_check) != 1){
+							fprintf(stderr, "Failed to set source"
+							" IPv6 address. Please check if"
+							" source is set to a valid IPv6 address.\n");
+							return 1;
+						}
+					}
+				}
+				else {
```

Comment:
Closing `{` and `else {` on the same line, please. I know this is not consistent across this file but while you're touching the code in question I think it's good to fix it up.